### ISSUE TITLE
pkg_tar support for pkg_filegroup/pkg_files with a tree artifact with subdirectories

### ISSUE THREAD

**Issue by brian-tecton-ai** [bazelbuild/rules_pkg#550]
It seems as though `pkg_tar` supports tree artifacts with subdirectories, but not when used in conjunction with a `pkg_filegroup`.

Repro BUILD.bazel (uses a copied version of the `directory` test util):

```
load("@rules_pkg_0_6//:pkg.bzl", "pkg_tar")
load("@rules_pkg_0_6//:mappings.bzl", "REMOVE_BASE_DIRECTORY", "pkg_attributes", "pkg_filegroup", "pkg_files", "pkg_mkdirs")
load("//test:helper.bzl", "directory")

directory(
    name = "test_dir1",
    contents = "test_dir1",
    filenames = [
        "a",
        "b",
        "subdir/c",
        "subdir/d",
    ],
    outdir = "test_dir",
)

pkg_files(
    name = "test_files",
    srcs = [":test_dir1"],
)

pkg_filegroup(
    name = "test_fg",
    srcs = [
        ":test_files",
    ],
)

pkg_tar(
    name = "test_tar",
    srcs = [":test_fg"],
)

pkg_tar(
    name = "test_tar_2",
    srcs = [":test_dir1"],
    package_dir="test_files_2",
)
```

After building the tars I dumped them both to a `/tmp/tar` and unarachived them:

```
>> ls
test_tar.tar  test_tar_2.tar

>> tar xvf test_tar_2.tar
x ./
x ./test_files_2/
x ./test_files_2/a
x ./test_files_2/b
x ./test_files_2/subdir/
x ./test_files_2/subdir/c
x ./test_files_2/subdir/d

>> tar xvf test_tar.tar
x ./
x ./test_dir/
x ./test_dir/a
x ./test_dir/b
x ./test_dir/subdir/
x ./test_dir/subdir/c
x ./test_dir/subdir/d

>> ls -al test_files_2/subdir
total 16
dr-xr-xr-x  4 brian  wheel  128 Dec 31  1999 .
drwxr-xr-x  5 brian  wheel  160 Dec 31  1999 ..
-r-xr-xr-x  1 brian  wheel    9 Dec 31  1999 c
-r-xr-xr-x  1 brian  wheel    9 Dec 31  1999 d

>> ls -al test_dir/subdir
total 0
```

As you see in the above output, the logged output of the `tar xvf` look the same (besides a diff prefix). However when you list the files in the subdirectory, the one that used a `pkg_filegroup` does not actually have the expected files.

System Info:
* MacOS 12.1 (Intel)
* bazel: 5.0.0rc3
* rules_pkg: 0.6.0

**Comment by anagrius** [comment]
I am seeing this issue as well.

**Comment by aiuto** [comment]
This seems fixed at head, and probably version 0.7.0.

The current behavior is as I suggested in the comment from 2022/02/24
